Gems/Atom/Asset/Shader/Code/Source/Editor/ShaderBuilderUtility.cpp

```diff
@@ -456,8 +457,9 @@ namespace AZ
                 const uint32_t rhiUniqueIndex, const AZStd::string& platformIdentifier, const AZStd::string& shaderJsonPath,
                 const uint32_t supervariantIndex, RPI::ShaderAssetSubId shaderAssetSubId)
             {
-                // platform id from identifier
-                AzFramework::PlatformId platformId = AzFramework::PlatformId::PC;
+                // Define a fallback platform ID based on the current host platform
+                AzFramework::PlatformId platformId = AZ_TRAIT_ATOM_FALLBACK_ASSET_HOST_PLATFORM;
+
                 if (platformIdentifier == "pc")
```

Comment:
What an unfortunate implementation - how do new platforms add to this without modifying atom?

A better way would be to pass the BuilderSDK PlatformInfo in, rather than the string name of the platform.  All BuilderSDK calls (CreateJob, ProcessJob) always include the platforminfo.  The platformInfo has tags to use.

Not saying we have to change this now as part of THIS edit but its something that needs fixing if we expect people to be able to transparently add new platforms via PAL.

Reply:
In particular, since it didn't handle "server" (which has been a platform for a long time now, not sure why it wasnt here), it would fallback to "pc", and some assets used the platform cache folder to reference dependent assets, which it wasnt able to find. (See the original error in the description)